utilities/blob_db/blob_db_impl.h

```diff
@@ -408,6 +411,23 @@ class BlobDBImpl : public BlobDB {
 
   std::list<std::shared_ptr<BlobFile>> obsolete_files_;
 
+  // DeleteObsoleteFiles, DiableFileDeletions and EnableFileDeletions block
+  // on the mutex to avoid contention.
+  //
+  // While DeleteObsoleteFiles hold both mutex_ and delete_file_mutex_, note
+  // the difference. mutex_ only needs to be held when access the
+  // data-structure, and delete_file_mutex_ needs to be held the whole time
+  // during DeleteObsoleteFiles to avoid being run simultaneously with
+  // DisableFileDeletions.
```

Comment:
This is not what I mean by holding order. By holding order, you need to explicitly say, which one needs to be hold before which one if they are to be hold together. An out-of-order holding in the future can cause deadlock.

Reply:
I mean, there isn't an ordering requirement between these mutex, since `DeleteObsoleteFiles` is the only guy that will hold both, and it is supposed to be run from 1 thread.

Reply:
It doesn't matter. The comment is for the future when someone might try to hold the two locks together in another function.

Reply:
Interesting topic. Agree that if we can define a total order to the locks, deadlock can be avoid. I'll add the comment anyway, but it still confuse me whether such ordering is needed, and whether we can always define such an order.

Reply:
I think of it a bit and realize acyclic dependency between the locks is a requirement. What confuse me was there isn't strict requirement between the two locks in this example, by the definition of the data-structure. But in my implementation I happened to hold `delete_file_mutex_` before `mutex_`, thus defining an order.